stdlib/public/core/SmallString.swift

```diff
@@ -243,7 +243,9 @@ extension _SmallString {
   fileprivate mutating func withMutableCapacity(
     _ f: (UnsafeMutableRawBufferPointer) throws -> Int
   ) rethrows {
-    let len = try withUnsafeMutableBytes(of: &_storage, f)
+    let len = try withUnsafeMutableBytes(of: &_storage) {
+      try f(.init(rebasing: $0.prefix(_SmallString.capacity)))
```

Comment:
It would be justifiable to use `.init(_uncheckedStart:count:)` here: the preconditions are known to be met.